src/mp4box/stsd.rs

```diff
@@ -51,6 +53,125 @@ impl Default for StsdBoxContent {
     }
 }
 
+impl StsdBoxContent {
+    /// Usually 8, but 10 for HDR (for example).
+    pub fn bit_depth(&self) -> Option<u8> {
+        #[allow(clippy::match_same_arms)]
+        match self {
+            Self::Av01(bx) => Some(bx.av1c.bit_depth),
+
+            Self::Avc1(_) => None, // TODO(emilk): figure out bit depth
+
+            Self::Hvc1(_) => None, // TODO(emilk): figure out bit depth
+
+            Self::Hev1(_) => None, // TODO(emilk): figure out bit depth
+
+            Self::Vp08(bx) => Some(bx.vpcc.bit_depth),
+
+            Self::Vp09(bx) => Some(bx.vpcc.bit_depth),
+
+            Self::Mp4a(_) | Self::Tx3g(_) | Self::Unknown(_) => None, // Not applicable
+        }
+    }
+
+    pub fn codec_string(&self) -> Option<String> {
+        Some(match self {
+            Self::Av01(Av01Box { av1c, .. }) => {
+                let profile = av1c.profile;
+                let level = av1c.level;
+                let tier = if av1c.tier == 0 { "M" } else { "H" };
+                let bit_depth = av1c.bit_depth;
+
+                format!("av01.{profile}.{level:02}{tier}.{bit_depth:02}")
+            }
+
+            Self::Avc1(Avc1Box { avcc, .. }) => {
+                let profile = avcc.avc_profile_indication;
+                let constraint = avcc.profile_compatibility;
+                let level = avcc.avc_level_indication;
+
+                format!("avc1.{profile:02X}{constraint:02X}{level:02X}")
```

Comment:
Just because I stumbled upon it, it's worth pointing out that these strings have special meaning

```suggestion
                // https://aomediacodec.github.io/av1-isobmff/#codecsparam
                format!("avc1.{profile:02X}{constraint:02X}{level:02X}")
```